src/GraphQL/Internal/Validation.hs

```diff
@@ -926,3 +928,29 @@ instance Applicative (Validator e) where
   Validator (Left e) <*> _ = Validator (Left e)
   Validator _ <*> (Validator (Left e)) = Validator (Left e)
   Validator (Right f) <*> Validator (Right x) = Validator (Right (f x))
+
+data DefinitionType = QueryDefinition | MutationDefinition deriving (Eq, Show)
+
+getDefinitionType :: AST.QueryDocument -> Maybe Name -> Maybe DefinitionType
```

Comment:
Would you mind adding a comment explaining a bit more about this function does? E.g. I'm not super clear on why `DefinitionType` needs to be wrapped in a `Maybe`.